packages/manager/apps/container/src/container/legacy/server-sidebar/universe/public-cloud/pci-menu.ts

```diff
@@ -49,382 +48,447 @@ export const features = [
   'public-cloud:project-settings',
   'ai-dashboard',
   'data-platform',
-  'ai-endpoints'
+  'ai-endpoints',
+  'key-management-service'
 ];
 
 export function getPciProjectMenu(
   projectId: string,
   region: string,
+  featureAvailabilities: Record<string, boolean>,
   getURL: (app: string, path: string) => string,
 ): PciMenuItem[] {
-  return [
-    {
+  const pciMenu: PciMenuItem[] = [];
+
+  const isFeaturesAvailable = (...features: string[]): boolean =>
+    Boolean(features.find((f: string) => featureAvailabilities[f]));
+
+  if (isFeaturesAvailable('instance')) {
+    pciMenu.push({
       id: 'compute',
       title: 'Compute',
       subItems: [
         {
           id: 'instance',
           title: 'Instances',
-          feature: 'instance',
           href: getURL('public-cloud', `#/pci/projects/${projectId}/instances`),
         },
       ],
-    },
-    {
+    });
+  }
+
+  if (
+    isFeaturesAvailable(
+      'block-storage',
+      'object-storage',
+      'archive',
+      'cold-archive',
+      'snapshot',
+      'volume-backup',
+      'instance-backup',
+      'workflow-management',
+    )
+  ) {
+    pciMenu.push({
       id: 'storage',
       title: 'Storage',
       subItems: [
-        {
+        isFeaturesAvailable('block-storage') && {
           id: 'block-storage',
           title: 'Block Storage',
-          feature: 'block-storage',
           href: getURL(
             'public-cloud',
             `#/pci/projects/${projectId}/storages/blocks`,
           ),
         },
-        {
+        isFeaturesAvailable('object-storage') && {
           id: 'object-storage',
           title: 'Object Storage',
-          feature: 'object-storage',
           href: getURL(
             'public-cloud',
             `#/pci/projects/${projectId}/storages/objects`,
           ),
         },
-        {
+         isFeaturesAvailable('archive') && {
```

Comment:
The result will be : [ {...item}, {...item}, false }.

It's wanted ?

Else we can delete the item directly :

```ts
        ...(isFeaturesAvailable('block-storage') ? {
          id: 'block-storage',
          title: 'Block Storage',
          href: getURL(
            'public-cloud',
            `#/pci/projects/${projectId}/storages/blocks`,
          ),
        } : {}),
        ...(isFeaturesAvailable('object-storage') ? {
          id: 'object-storage',
          title: 'Object Storage',
          href: getURL(
            'public-cloud',
            `#/pci/projects/${projectId}/storages/objects`,
          ),
        } : {}),
         ...(isFeaturesAvailable('archive') ? {
```

Reply:
The goal is to be compliant with the other menus, that's why I've used this syntax :+1: